pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -76,11 +77,79 @@ func (pl *CSILimits) Name() string {
 // failed by this plugin schedulable.
 func (pl *CSILimits) EventsToRegister() []framework.ClusterEventWithHint {
 	return []framework.ClusterEventWithHint{
-		{Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+        {Event: framework.ClusterEvent{Resource: framework.CSINode, ActionType: framework.Add}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
 	}
 }
 
+func (pl *CSILimits) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.QueueSkip, fmt.Errorf("unexpected objects in isSchedulableAfterPodDeleted: %w", err)
+	}
+
+	if oldPod.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	if len(oldPod.Spec.Volumes) == 0 {
+		return framework.QueueSkip, nil
+	}
+
+	csiNode, err := pl.csiNodeLister.Get(oldPod.Spec.NodeName)
+	if err != nil {
+		return framework.QueueSkip, fmt.Errorf("could not get a CSINode object for: %w", err)
+	}
+
+	oldPodDrivers := []string{}
```

Comment:
change it `sets.Set[string]` then it'd be easier to look up the volume from it instead of for-loop in L143